src/store/transactions/post-conditions.hooks.ts

```diff
@@ -1,6 +1,28 @@
-import { postConditionModeState } from '@store/transactions/post-conditions';
 import { useAtomValue } from 'jotai/utils';
+import { addressToString, PostCondition, PostConditionType } from '@stacks/transactions';
+import { postConditionModeState } from '@store/transactions/post-conditions';
+import { useFungibleTokenMetaDataState } from '@store/assets/fungile-tokens.hooks';
 
 export const usePostConditionModeState = () => {
   return useAtomValue(postConditionModeState);
 };
+
+export const useAssetFromPostCondition = (pc: PostCondition) => {
+  let contractAddress: string = '';
+  let contractName: string = '';
+  if (pc.conditionType === PostConditionType.Fungible) {
+    contractAddress = addressToString(pc?.assetInfo.address);
+    contractName = pc.assetInfo.contractName.content;
+  }
+
+  const contractId = `${contractAddress}.${contractName}`;
+  const asset = useFungibleTokenMetaDataState(contractId);
+
+  if (!asset) return;
+  // Need to check this here bc the contractId can be '.' and return
+  // an object with all undefined property values
+  const isEmpty = Object.values(asset).every(x => x === undefined);
+  if (isEmpty) return;
+  if ('error' in asset) return;
+  return asset;
+};
```

Comment:
I find this hook quite hard to read. 

If `pc.conditionType` is something other than `PostConditionType.Fungible`, we pass `"."` to ` the hook, and rely on the fact it will return nothing. We could be more explicit about this error case behaviour.

If the function should only work for `PostConditionType.Fungible` should we name it as such, and return early if not the case? Then we don't have to declare empty strings and redefine them.

```suggestion
export const useAssetFromFungiblePostCondition = (pc: PostCondition) => {
  if (pc.conditionType !== PostConditionType.Fungible) return; // or throw an error?

  const contractAddress = addressToString(pc?.assetInfo.address);
  const contractName = pc.assetInfo.contractName.content;

  const contractId = `${contractAddress}.${contractName}`;
  const asset = useFungibleTokenMetaDataState(contractId);

  if (!asset) return;
  // Need to check this here bc the contractId can be '.' and return
  // an object with all undefined property values
  const assetOnlyHaveUndefinedProperties = Object.values(asset).every(x => x === undefined);
  if (assetOnlyHaveUndefinedProperties) return;
  if ('error' in asset) return;
  return asset;
};
```

I don't really get the isEmpty check. If `useFungibleTokenMetaDataState` doesn't find a contract, it returns an asset object but with all the properties set to `undefined`? 🤯  Surely not, or is there something else going on 🤔

Reply:
So, this was refactor to address @aulneau's comment that I was breaking the rules of hooks by returning early on the conitional `if (pc.conditionType !== PostConditionType.Fungible) return;`. Now, it does do the fetch and uses the `.` contractId. It does return an object w/ all property values undefined. It is the only way I could think of to catch it after the hook and return. Any better ideas? The NFT post conditions do fail it I don't return early.

Reply:
I see your comment about breaking the rule, I'll try to refactor to a better solution. I didn't write this originally so maybe I can catch it before it gets here and only use if a FT.

Reply:
A common solution is to remove the conditional from the hook entirely, and pushing it to the component level, this is an example:
```ts
export const useAssetFromFungiblePostCondition = (pc: PostConditionFungible) => {
  const contractAddress = addressToString(pc.assetInfo.address);
  const contractName = pc.assetInfo.contractName.content;
  const contractId = `${contractAddress}.${contractName}`;
  const asset = useFungibleTokenMetaDataState(contractId);
  return !asset || 'error' in asset ? undefined : asset
};
```

and then in whatever component it is, you'd do some kind of condition check:

```tsx
const FungiblePostCondition = ({pc}: {pc:PostConditionFungible}) => {
  const asset = useAssetFromFungiblePostCondition(pc)
  // the rest of the component
}

const PostConditionComponent = ({pc}:{pc: PostCondition}) => {
  if (pc.conditionType == PostConditionType.Fungible) return <FungiblePostCondition pc={pc} />;
  // the other post condition types here
  // <NonFungiblePostCondition />
  // <StxPostCondition />
}

Reply:
I did try it that way first, but it never rendered the NFT post condition in the test-app. I think I wasn't familiar enough with the code to add the `<NonFungiblePostCondition />` separately. Will do that, thx. 👍